### ISSUE TITLE
No fallback images

### ISSUE THREAD

**Issue by chickenputty** [surgeplay/Visage#18]
If an image fails to return, it should return a Steve head on default. This would prevent having empty spaces were heads should go.

![image](https://cloud.githubusercontent.com/assets/1813460/24971071/3d664bfa-1f7c-11e7-9cc1-fd310adc0b91.png)

A bugged account where this would apply:
https://visage.surgeplay.com/head/48/8a279cc099f34aababc71701e7dafc24.png

And of course, this would apply to all invalid uuids or names

**Comment by SeruhioX** [comment]
Do you know anything about this?

I am trying to use this api and would need this function

thx

**Comment by exaskye** [comment]
The issue shown in the OP was a problem where one of the renderers would sometimes serve up blank images. That's since been fixed, but if you actually mean the intent of the issue, where an *invalid* uuid serves a fallback image, I'm against this behavior as it **is** an error condition to try to render a skin for a nonexistant account.

That said, I'm not against adding this as opt-in, e.g. ?fallback={steve,alex,adaptive}

Also note that username renders are *unsupported* and soon will serve up images like this one: 
<img src="https://user-images.githubusercontent.com/6185037/30093818-ee665f44-9295-11e7-81d2-7ea62851b0b8.png" width="240px"/>

So if your reason for wanting this is people typo'ing their name, what you're doing is bad and unsupported and you should stop. Or convince me to not do this.

If your reasoning is, instead, to support the fake UUIDs generated for offline-mode, then I can make Visage detect those (they're a different kind of UUID than normal ones) and serve up a default skin based on if the UUID is odd or even.

**Comment by SeruhioX** [comment]
I'm using litebans, and for some reason non-premium users do not show them the uuid if not the username

Adding something like "?fallback=X-Steve" would be perfect

Thank you for your work and quick response.

**Comment by exaskye** [comment]
If by "non-premium" you mean Minecraft accounts, then as mentioned, I can add specific support for offline mode. I'd consider offline mode username renders to even be *supported* (!) because what username turns into what skin is deterministic (based on MD5). Does a query like `?offline-mode=true` also sound useful to enable this behavior?

**Comment by SeruhioX** [comment]
I imagine a query like https://visage.surgeplay.com/head/48/nickinvented?offline-mode=true

And return an image instead of error 400, it would be fine (if that's how I think it works)

I'm not very sure since my server uses uuid both online and offline

**Comment by exaskye** [comment]
Yeah, it would redirect to https://visage.surgeplay.com/head/48/5cb26ca5b88d3357857c218763b39777 which is a Version 3 UUID (i.e. offline mode - premium accounts are Version 4) and would return either Alex or Steve.

**Comment by SeruhioX** [comment]
So would it look like this using both online and offline uuid?

To be specific, I use a plugin that converts legitimate users to uuid online, but I also have uuid offline, is what I am not clear, with this, I would not show the skin of users also if they have an online uuid?

**Comment by exaskye** [comment]
Player UUIDs have a "version" (the bold part) - 5cb26ca5b88d**3**357857c218763b39777  
Since that's a 3, you *know* it's from offline-mode. Visage can detect this and give you a default skin always.

To compare, my UUID is e88b3493ea6b**4**2dc94ce1bdfda76ebbd  
Since it's a 4, you *know* it's a real account in online-mode. Visage can detect this and download the skin from Mojang.

The ?offline-mode=true option would simply convert usernames into version 3 UUIDs just like how vanilla does it.

**Comment by SeruhioX** [comment]
Then it would be great if you could implement it

Thanks for your time
